### DISCUSSION TITLE
Tutorial for MCC

### DISCUSSION THREAD

**Discussion by Daenges** [ORelio/Minecraft-Console-Client#1903] [Show and tell]
Hello there. When I [searched on Youtube](https://www.youtube.com/results?search_query=Minecraft+Console+Client) for videos on MCC, I only found ones, that not even mention this repository, provide a zipped version (from Mediafire) with some custom scripts and rush through the setup.

Therefore I started my own tutorial series. My [first video](https://youtu.be/BkCqOCa2uQw) just launched.
I am happy about **every feedback I will get**. :)

**Comment by ORelio** [comment]
Congrats! Visually it's great, well animated and explained.
The generated voice feels a bit unnatural but also quite easy to understand so it's OK.

Regarding what is presented in the video, maybe it should have started by what is it before explaining where to ask questions, as well as "can you trust" it, these look more like questions that could be answered later in the video 😉

What is MCC?
What is it used for?
How can you download MCC?
How can you start MCC?
Can you trust MCC?
Questions?

I'd suggest an order like this, maybe.
But anyway thanks for working on this 👍

**Reply by Daenges** [reply]
Thank you for your feedback!

I chose this layout, because **NO** other video about this client links to this Github page. Therefore I wanted to catch the viewer straight from the beginning and try to get rid of all misinformation. (Some videos create a picture, that suggests that the reviewing Youtuber is responsible for MCC. Even their comments contain some support questions.)
As a result it was important for me to correct this for everyone, right in the beginning, even if they turn the video off after a minute.

I am currently planning a video about Commands, Scripts and some triggers for scripts. I will try to use a clearer structure here. :)

**Reply by ORelio** [reply]
>NO other video about this client links to this Github page

Indeed. That's mainly why I ended up adding "Github.com/MCCTeam" in the first line shown in the program. Anyone curious enough will find the repository. Of course this does not protect against _modified versions_ that would remove the credits 😉

> I am currently planning a video about Commands, Scripts and some triggers for scripts. I will try to use a clearer structure here. :)

Great! Don't forget to have fun making it! 😁

**Comment by Daenges** [comment]
Hello There. My [second video](https://youtu.be/JbDpwwETEnU) about commands and scripting is now on Youtube.
It addresses new users and should help them to cope with typical issues while they are scripting for the first time. 
The video took longer than expected, since I found it pretty difficult to sum everything up, while staying as detailed and close to the original documentation as possible and providing visually appealing examples.

Again I will be happy about any feedback! Since you were the only one the last time, you get a ping: @ORelio :P

**Comment by osfanbuff63** [comment]
Sounds great! Maybe for text-to-speech, you could use [this from IBM](https://www.ibm.com/demos/live/tts-demo/self-service/home) (you'd have to record it using OBS or something, unless you sign up for an account), since it sounds a bit more realistic I think.

**Reply by Daenges** [reply]
Thanks for the tip! Currently I am using [this site](https://freetts.com/) for the TTS voice. Here you do not need an account and you can directly export the spoken scripts in the `.mp3` format. This makes life way easier at the price of a less realistic voice.

**Comment by Sasubaysa** [comment]


**Reply by Daenges** [reply]
What's the Problem?

**Reply by Daenges** [reply]
Send the script and explain what you want to achieve with it.

**Reply by Sasubaysa** [reply]
@Daenges  
First of all, there are minions on the server I play on and we can feed the minions by right-clicking.  Is it possible to do something like this in MCC?

**Reply by Sasubaysa** [reply]
If we right-click on the minion while we have a steak, the minion's health is restored l

**Reply by Sasubaysa** [reply]
Can we do something like this on console?

**Reply by Daenges** [reply]
You can use /inventory and /entity commands for this

**Reply by Sasubaysa** [reply]
Minions consist of item hangers.  so not alive

**Reply by Sasubaysa** [reply]
Not working

**Reply by Daenges** [reply]
You mean item frames?
Those should be entities.

**Reply by Sasubaysa** [reply]
goods hanger

**Reply by Sasubaysa** [reply]
![IMG_20230329_124736.jpg](https://user-images.githubusercontent.com/129219860/228495772-1432e4a7-e670-48ba-a66b-a10e800e5161.jpg)

**Reply by Daenges** [reply]
This is an Armor Stand. Also an entity.

**Reply by Sasubaysa** [reply]
what exactly should i do ?

**Reply by Daenges** [reply]
- Activate Inventory Handling
- Activate Entoty Handling

Use `/help inventory` or `/help entity` to gett all available commands.
Put the desired item in your hotbar.

Select this slot with /inventory and perform a rightclick on the entity with /entity.

**Reply by Sasubaysa** [reply]
[MCC] It says 2 entities are used but does not feed the minion

**Reply by Sasubaysa** [reply]
![IMG_20230329_134638.jpg](https://user-images.githubusercontent.com/129219860/228510401-981aa7bc-f220-4a08-9a81-554d4d123474.jpg)

**Reply by Sasubaysa** [reply]
we took the steak and used the command /entity ArmorStand use, the result did not change

**Reply by Daenges** [reply]
Uhh.. then i need to test it later.

**Reply by breadbyte** [reply]
From your screenshot, it looks like the bot is holding the steak in the offhand. Equip it in the main hand, select it with `/inventory`, and then try again.

**Reply by Sasubaysa** [reply]
@breadbyte  When we enter it from the computer, it appears in the main hand, but when we enter the console, it appears in the 2nd hand. What should I do?

**Reply by Daenges** [reply]
The default main hand setting is left. So it should be in main Hand.

**Reply by Sasubaysa** [reply]
@breadbyte  
Also, I don't quite understand the select with /inventory part.  Isn't it enough to take the cow steak as the main?

**Reply by Sasubaysa** [reply]
I set the main hand to right in Minecraft.ini.

**Reply by breadbyte** [reply]
> @breadbyte Also, I don't quite understand the select with /inventory part. Isn't it enough to take the cow steak as the main?

You need to use `/inventory` to change the hotbar slot (like pressing 1-9 on your keyboard to change slot.)

**Reply by Sasubaysa** [reply]
@breadbyte  ![IMG_20230329_143110.jpg](https://user-images.githubusercontent.com/129219860/228522664-2da886b0-63ea-4d5d-8657-6968df51d26c.jpg)

![IMG_20230329_143115.jpg](https://user-images.githubusercontent.com/129219860/228522679-0e4fc397-704d-4fc8-8bdc-63feb5175295.jpg)



5 entity kullanıldı = 5 entities used

**Comment by MousePak** [comment]
@Sasubaysa The minions might be a replaced blocks in the game. For example if you want to interact with grindstone you first need to open its gui (within mcc they are \inventory container list). Daenges has a lot of knowledge so wont cover whats already been suggested.

try using the block (minion) coordinates
\useblock x y z 

If lucky you will have a new inventory created you could then interact with

-------
While testing armourstand you will need to use \entity 
[MCC] Please enable EntityHandling in the config file first.
EntityHandling = true

\entity near

[MCC] Entity: 1372085
 [MCC] Type: Armor Stand
 [MCC] Pose: Standing
 [MCC] Health: 20
 [MCC] Distance: 1.41
 [MCC] Location: X:1.5, Y:60, Z:0.5
 
 if you have this so far... I need further testing but get the food in hand on the bot .. sign off and back to console then \entity use should use the item in hand with the stand.

on a default armor stand holding leggings ive been unable to put the on the stand how I expect this to work !

update: 
while holding leggings in hand
\entity 12345 use 
\entity 12345 attack
^ both do not put the leggings on the stand ... this might be a bug, could someone confirm ?

**Reply by Sasubaysa** [reply]
It doesn't work, is there a way to open the minion's gui?

**Reply by MousePak** [reply]
Im a little stumped so logged a bug / feature request as the stand isnt working as expected 
> https://github.com/MCCTeam/Minecraft-Console-Client/issues/2463

**Reply by Sasubaysa** [reply]
Thx
